jerry-core/ecma/builtin-objects/ecma-builtin-object-prototype.c

```diff
@@ -143,7 +144,23 @@ static ecma_value_t
 ecma_builtin_object_prototype_object_has_own_property (ecma_object_t *obj_p, /**< this argument */
                                                        ecma_string_t *prop_name_p) /**< first argument */
 {
-  return ecma_make_boolean_value (ecma_op_object_has_own_property (obj_p, prop_name_p));
+#if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
+  if (ecma_object_is_proxy (obj_p))
+  {
+    ecma_property_descriptor_t prop_desc;
+
+    ecma_value_t status = ecma_proxy_object_get_own_property_descriptor (obj_p, prop_name_p, &prop_desc);
```

Comment:
Would it make sense to pass `NULL` as the third argument indicating that we don't want any data back? If so we could spare the free call below.

Reply:
To be honest, I think the users 95% do not know about the existence of this function (the remaining 5% are JS engine developers :) ). The `ecma_proxy_object_get_own_property_descriptor` is a frequently used function, where you should fill all the structure's field with the current property data if the property exists. So i'd not burden with extra null checks the general proxy method due to this case where the descriptor's data is not needed at all.